lib/_tls_legacy.js

```diff
@@ -585,14 +585,12 @@ function onhandshakestart() {
   ssl.lastHandshakeTime = now;
   if (first) return;
 
-  if (++ssl.handshakes > tls.CLIENT_RENEG_LIMIT) {
+  if (++ssl.handshakes > tls.CLIENT_RENEG_LIMIT && self.cleartext) {
```

Comment:
It seems to me that it's likely inside the `setImmediate()` because it could potentially change by that point. I'd also be very wary changing `setImmediate()` to `process.nextTick()` for this one. Needs input from @nodejs/crypto I think.